### PR TITLE
HPCC-28961 Members of Regression Suite spray class generate and use incorrect temporary logical and physical file paths.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Fix despray path generation for:
- despray.ecl
- spray_header_test.ecl
- spray_test_json.ecl
- spray_test_xml.ecl

The despray.ecl extended with test to despray into a (Drop Zone) relative path.

Except the split_test.ecl all Spray class member exclude from Hthor and Thor engines.

The split_test.ecl run only on Thor.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested manually on local VM.

### CODE REVIEW

File: testing/regress/ecl/despray.ecl [hunk 1]
```diff
@@ -232,7 +232,7 @@ c8 := CATCH(NOFOLD(p8), ONFAIL(TRANSFORM(rec,
 
 
 // This should pass based on valid target file path and valid source address used
-DestFile9 := dropzonePath + 'test/' + prefix + File;
+DestFile9 := dropzonePath + 'test/' + WORKUNIT + '-' + File;
```
Comment (jakesmith) [inline testing/regress/ecl/despray.ecl:235]: minor: for consistency should use 'DestFileName' here instead WORKUNIT + '-' + File

NB: line 192, 214 should be changed too.

File: testing/regress/ecl/despray.ecl [hunk 2]
```diff
@@ -274,6 +274,29 @@ c10 := CATCH(NOFOLD(p10), ONFAIL(TRANSFORM(rec,
     o10 := output(c10, {result});
 #end
 
+
+
+// This should pass based on valid relative file path and valid source address used
+DestFile11 := 'test/' + WORKUNIT + '-' + File;
```
Comment (jakesmith) [inline testing/regress/ecl/despray.ecl:280]: minor: as per previous comment, for consistency should use DestFileName here.

**Review by jakesmith — APPROVED** [review]
@AttilaVamos - looks good, could be merged as is, but 1 minor issue you might want to address 1st.

File: testing/regress/ecl/despray.ecl (hunk 1 above)
Comment (AttilaVamos) [reply testing/regress/ecl/despray.ecl]: Changed

File: testing/regress/ecl/despray.ecl (hunk 2 above)
Comment (AttilaVamos) [reply testing/regress/ecl/despray.ecl]: Changed

**Comment by ghalliday** [comment]
@AttilaVamos merging as-is.  It would be better if the following were in a shared file (e.g. setup/options.ecl) so they could be reused by all the tests:
```
dropzonePathTemp := '/var/lib/HPCCSystems/mydropzone/' : STORED('dropzonePath');
dropzonePath := dropzonePathTemp + IF(dropzonePathTemp[LENGTH(dropzonePathTemp)]='/', '', '/');
```

**Pull request merged**